### PR TITLE
Python3

### PR DESCRIPTION
**Pull request by devang-dreambits**
Porting python bol api to python3

### CODE REVIEW

File: bol/plaza/api.py [hunk 1]
```diff
@@ -148,7 +147,7 @@ def create_request_xml_elements_for_create_inbound(self, indent, **kwargs):
         # sort to make output deterministic
         kwargs = collections.OrderedDict(sorted(kwargs.items()))
         xml = ''
-        for tag, value in kwargs.items():
+        for tag, value in list(kwargs.items()):
```
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: The tools itself has made me to do this...

```
 # Dynamically calculate the version based on actistream.VERSION.
 VERSION = bol.__version__
--- ./bol/plaza/api.py	(original)
+++ ./bol/plaza/api.py	(refactored)
@@ -147,7 +147,7 @@
         # sort to make output deterministic
         kwargs = collections.OrderedDict(sorted(kwargs.items()))
         xml = ''
-        for tag, value in kwargs.iteritems():
+        for tag, value in kwargs.items():
             if value is not None:
                 prefix = ' ' * 4 * indent
                 if not isinstance(value, list):--- ./__init__.py	(original)
```

------------------------------------------
after 2nd run 

```
 # Dynamically calculate the version based on actistream.VERSION.
 VERSION = bol.__version__
--- ./bol/plaza/api.py	(original)
+++ ./bol/plaza/api.py	(refactored)
@@ -147,7 +147,7 @@
         # sort to make output deterministic
         kwargs = collections.OrderedDict(sorted(kwargs.items()))
         xml = ''
-        for tag, value in kwargs.items():
+        for tag, value in list(kwargs.items()):
             if value is not None:
                 prefix = ' ' * 4 * indent
                 if not isinstance(value, list):
```

File: bol/plaza/api.py [hunk 2]
```diff
@@ -459,7 +458,7 @@ def deleteOffers(self, offers, path='/', params={},
             if response is True:
                 return response
         except Exception:
-            print("Got into Exception \n{0}".format(traceback.print_exc()))
+            print(("Got into Exception \n{0}".format(traceback.print_exc())))
```
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: yes, buy what I had observed is, tool 2to3 doesn't notify any change if we print any object directly

```
print(xml)
```
But, when only ask for double bracket when we are formatting the string and printing with var or objects.

File: bol/plaza/api.py [hunk 3]
```diff
@@ -553,9 +552,9 @@ def check_prod(self, prod):
         if not isinstance(prod['Product'], dict):
             type_exception('dict', prod['Product'])
 
-        if 'EAN' not in prod['Product'].keys():
+        if 'EAN' not in list(prod['Product'].keys()):
```
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: Maybe the type of the 'keys()' has been changed in pytho3 and so it is asking to make it list for that loop...

$ python3
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {'a':1,'b':2}
>>> a.keys()
dict_keys(['a', 'b'])
>>> list(a.keys())
['a', 'b']
>>> 

and in python2 

$ python2
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {'a':1,'b':2}
>>> a.keys()
['a', 'b']
>>>

File: bol/plaza/api.py [hunk 4]
```diff
@@ -553,9 +552,9 @@ def check_prod(self, prod):
...
             key_exception('EAN')
-        if 'AnnouncedQuantity' not in prod['Product'].keys():
+        if 'AnnouncedQuantity' not in prod_keys:
```
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: I had maintained py2 and py3 compatibility as we did for DHL plugin and but later I removed py2  so this was I forget to replace.

File: bol/plaza/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (karan-dreambits) [reply bol/plaza/api.py]: so you run 2to3 tool two times ?

File: bol/plaza/api.py [hunk 5]
```diff
@@ -198,7 +197,7 @@ def _create_request_xml_elements(self, indent, **kwargs):
         # sort to make output deterministic
         kwargs = collections.OrderedDict(sorted(kwargs.items()))
         xml = ''
-        for tag, value in kwargs.items():
+        for tag, value in list(kwargs.items()):
```
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: replied to above.

File: bol/plaza/api.py [hunk 6]
```diff
@@ -721,5 +720,5 @@ def request(self, method, uri, params={},
                 tree = ElementTree.fromstring(resp.content)
                 return tree
         except Exception:
-            print("Got into Exception \n{0}".format(traceback.print_exc()))
+            print(("Got into Exception \n{0}".format(traceback.print_exc())))
```
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: replied to above.

File: bol/plaza/api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (karan-dreambits) [reply bol/plaza/api.py]: so this needs to be changed, right ?

File: bol/plaza/api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (karan-dreambits) [reply bol/plaza/api.py]: yes, you are right that keys() method has changed in python 3 :

https://docs.python.org/3/whatsnew/3.0.html#views-and-iterators-instead-of-lists

But I think 'EAN' not in prod['Product'].keys() should work fine.
You should check that

File: bol/plaza/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: yes to see there is nothing left, a few of the repetitive suggestion I have changed by finding all and replace all, so I run more than 4-5 times,...

File: bol/plaza/api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: yes, I changed in my local.

File: bol/plaza/api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (devang-dreambits) [reply bol/plaza/api.py]: in that case, I need to manually check for this kind of performance points, ... because 2to3 tool is just formatting to syntax.

**Pull request merged**